solc/CommandLineInterface.cpp

```diff
@@ -574,6 +585,7 @@ bool CommandLineInterface::processInput()
 	};
 
 	m_compiler.reset(new CompilerStack(m_args.count(g_argAddStandard) > 0, fileReader));
+	auto scannerFromSourceName = [&](string const& _sourceName) -> solidity::Scanner const& { return m_compiler->scanner(_sourceName); };
```

Comment:
correct me if I'm wrong Chris, the lambda function captures objects by copy, they can't be modified inside(->), and returns const& to scanner object based on _sourceName.